packages/typespec-vscode/src/vscode-cmd/emit-code/emit-code.ts

```diff
@@ -0,0 +1,337 @@
+      emitterKind: e.kind,
+      label: e.language,
+      detail: `Generate ${e.kind} code for ${e.language} by TypeSpec library ${e.package}.${moreDetail}`,
+      picked: false,
+      fromConfig: false,
+      iconPath: Uri.file(
+        context.asAbsolutePath(`./icons/${getLanguageAlias(e.language).toLowerCase()}.svg`),
+      ),
+    };
+  };
+
+  const registerEmitters = getRegisterEmitters(kind);
+  const all = [...registerEmitters].map((e) => toQuickPickItem(e));
+
+  const selectedEmitter = await vscode.window.showQuickPick<EmitQuickPickItem>(all, {
+    title: `Select a Language for ${kind} code generation`,
+    canPickMany: false,
+    placeHolder: `Select a Language for ${kind} code generation`,
+    ignoreFocusOut: true,
+  });
+
+  if (!selectedEmitter) {
+    logger.info("No language selected. Generating Cancelled.");
+    return;
+  }
+
+  const npmUtil = new NpmUtil(baseDir);
+  const packagesToInstall: string[] = [];
+
+  /* install emitter package. */
+  logger.info(`select ${selectedEmitter.package}`);
+  const { action, version } = await npmUtil.calculateNpmPackageInstallAction(
+    selectedEmitter.package,
+    selectedEmitter.version,
+  );
+
+  if (action === InstallationAction.Upgrade) {
+    logger.info(`Upgrading ${selectedEmitter.package} to version ${version}`);
+    const options = {
+      ok: `OK (install ${selectedEmitter.package}@${version} by 'npm install'`,
+      recheck: `Check again (install ${selectedEmitter.package} manually)`,
+      ignore: `Ignore (don't upgrade emitter ${selectedEmitter.package})`,
+    };
+    const selected = await vscode.window.showQuickPick(Object.values(options), {
+      canPickMany: false,
+      ignoreFocusOut: true,
+      placeHolder: `Package '${selectedEmitter.package}' needs to be upgraded for generating`,
+      title: `TypeSpec Generating...`,
+    });
+    if (selected === options.ok) {
+      packagesToInstall.push(`${selectedEmitter.package}@${version}`);
+    } else if (selected === options.ignore) {
+      logger.info(`Ignore upgrading emitter ${selectedEmitter.package} for generating`);
+    } else {
+      logger.info(
+        `Need to manually install the package ${selectedEmitter.package}@${version}. Generating Cancelled.`,
+        [],
+        {
+          showOutput: false,
+          showPopup: true,
+        },
+      );
+      return;
+    }
+  } else if (action === InstallationAction.Install) {
+    let packageFullName = selectedEmitter.package;
+    if (version) {
+      packageFullName = `${selectedEmitter.package}@${version}`;
+    }
+    logger.info(`To install ${packageFullName}`);
+    /* verify dependency packages. */
+    const dependenciesToInstall = await npmUtil.calculateNpmPackageDependencyToUpgrade(
+      selectedEmitter.package,
+      version,
+      npmDependencyType.peerDependencies,
+    );
+    logger.info(`${dependenciesToInstall}`);
+    if (dependenciesToInstall.length > 0) {
+      vscode.window.showInformationMessage(
```

Comment:
please also log the message, it's hard to read the long message in notification box. You can just use logger.info with showPopup = true